frontend/src/libs/components/router/router.tsx

```diff
@@ -1,34 +1,31 @@
+import { Route, Routes } from 'react-router-dom';
+
 import { AppRoute } from '~/libs/enums/enums.js';
 import { Auth } from '~/pages/auth/auth.js';
 import { NotFound } from '~/pages/not-found/not-found.js';
 
 import { App } from '../app/app.js';
+import { ProtectedRoute } from '../components.js';
 import { RouterProvider } from '../router-provider/router-provider.js';
 
 const Router = (): JSX.Element => (
   <RouterProvider
     routes={[
       {
-        path: AppRoute.ROOT,
-        element: <App />,
-        children: [
-          {
-            path: AppRoute.ROOT,
-            element: 'Root',
-          },
-          {
-            path: AppRoute.SIGN_IN,
-            element: <Auth />,
-          },
-          {
-            path: AppRoute.SIGN_UP,
-            element: <Auth />,
-          },
-          {
-            path: AppRoute.ANY,
-            element: <NotFound />,
-          },
-        ],
+        path: '*',
+        element: (
+          <Routes>
+            <Route path={AppRoute.ROOT} element={<App />}>
+              <Route path={AppRoute.SIGN_IN} element={<Auth />} />
+              <Route path={AppRoute.SIGN_UP} element={<Auth />} />
+            </Route>
+            <Route path={AppRoute.ROOT} element={<ProtectedRoute />}>
+              <Route path="protected1" element={<div>Protected path #1</div>} />
+              <Route path="protected2" element={<div>Protected path #2</div>} />
+            </Route>
+            <Route path={AppRoute.ANY} element={<NotFound />} />
+          </Routes>
```

Comment:
If ProtectedRoute needs to be nested in the App layout, I will move it there without any problems.
```suggestion
          <Routes>
            <Route path={AppRoute.ROOT} element={<App />}>
              <Route path={AppRoute.SIGN_IN} element={<Auth />} />
              <Route path={AppRoute.SIGN_UP} element={<Auth />} />
              <Route element={<ProtectedRoute />}>
                <Route
                  path="protected1"
                  element={<div>Protected path #1</div>}
                />
                <Route
                  path="protected2"
                  element={<div>Protected path #2</div>}
                />
              </Route>
            </Route>
            <Route path={AppRoute.ANY} element={<NotFound />} />
          </Routes>
```